### DISCUSSION TITLE
Programming pins

### DISCUSSION THREAD

**Discussion by moto25** [revk/xESP32-Faikout#367] [General]
Hi, what are the programming pins for this board? Thank you
![20240625_155024](https://github.com/revk/ESP32-Faikin/assets/99315732/2a327920-b82f-4f06-98cf-de91383e8d28)

**Comment by revk** [comment]
I have not shipped any with those yet :-)

They are USB - the idea was they could be used with 4 pins or 4 spring pins. I decided they don't work well as the pins can scratch the tracks on the back of the processor, so trying something different for current design.

![324013](https://github.com/revk/ESP32-Faikin/assets/996983/4918fee8-42d5-46fb-8c9b-8de6538ee831)

**Comment by moto25** [comment]
Ouh, i thought Fainkin is tested and Faikin+ isnt . I got them already so need to flash them somehow. What u recommend? Thanx

**Reply by revk** [reply]
Once upon a time, the Faikin+ is now a dual interface one (not updated for connectors yet).

That header does work, you just have to be careful.

I used https://www.amazon.co.uk/dp/B0CZ675CRD with bent pins.

**Comment by moto25** [comment]
Did my own cable,  are you sure the pins work properly? In logs i see just connect/disconnect loops, thanx

![20240628_151953](https://github.com/revk/ESP32-Faikin/assets/99315732/b6aa2ba7-a8fe-4beb-91ac-7497505562fb)
![20240628_151706](https://github.com/revk/ESP32-Faikin/assets/99315732/45694c74-9e65-47a0-ac07-3b38e611df81)

**Reply by revk** [reply]
The pins definitely work, the only issue I have had is the pins scraping the back of the processor and shorting, so you have to be rather careful.

**Reply by dwareb** [reply]
I had few boards manufactured for myself too, and am running into the same problem with the boot loop / it constantly disconnecting / reconnecting. 

I had a few extra made, because, I know myself and know that I'll generally break a couple while tinkering with them. I've tried a few things to get the USB header connected, including pins, much like you have above, and soldering wires directly to the pads. I'm positive I must have scraped the back of the processor with the first one I tried, so I have been quite careful since to make sure I don't do that again. But I'm still running into the disconnects/reconnects. I'm assuming the move towards the tag-connect ones for future versions is to help address this.

@moto25 were you able to figure anything out? I feel like I might just be missing something obvious...

Oh, and just for context, the board rev I had made was a little different than this one. It was the 2024-03-28 one.  
![faikin](https://github.com/revk/ESP32-Faikin/assets/53237662/d7c7036f-73b4-41a3-aaa2-d33e7bc86b06)

**Reply by dwareb** [reply]
Update, I think I've gotten it to work. It also doesn't look like I damaged any of the boards I was worried that I had.

What did I do? Well... spam the attempt to flash it. Since on my system (I'm by no means a linux expert... novice at best) wasn't giving me write privs on the device, I had to modify those for the device, and try to write at nearly the same time. So I just concatenated the commands together.

`sudo chmod a+rw /dev/ttyACM0 && make flash`

I tried this a few times, and that got it to work. Eventually, it'd catch it at the right time, and begin flashing the chip before it rebooted. I've done it a few times now, and it works pretty reliably for me... Now to mess around with the setup and see if I can wire these things into my AC. 

Btw, @revk you're a rockstar, and this project is tip-top.
